app/controllers/v0/messages_controller.rb

```
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+module V0
+  class MessagesController < HealthcareMessagingController
+    def index
+      folder_id = params[:folder_id].try(:to_i)
+      response = client.get_folder_messages(folder_id, 1, 100)
+
+      raise VA::API::Common::Exceptions::RecordNotFound, folder_id unless response.present?
+
+      render json: response.data,
+             serializer: CollectionSerializer,
+             each_serializer: MessageSerializer,
+             meta: response.metadata
+    end
+
+    def show
+      message_id = params[:id].try(:to_i)
+      response = client.get_message(message_id)
+
+      raise VA::API::Common::Exceptions::RecordNotFound, message_id unless response.present?
+
+      render json: response.data[0],
+             serializer: MessageSerializer,
+             meta: response.data[0].metadata
+    end
+
+    def create
+      subject = params[:subject]
+      category = params[:category]
+      recipient_id = params[:recipient_id]
+      body = params[:body]
+
+      response = client.post_create_message(subject: subject,
```

Could use the Internal server error in va common. Although I encounted a net timeout error when a valid recipient id of a triage team (one returned in the list of triage teams) was nevertheless unable to accept a new message. The true nature of the error would be hidden.